pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -2289,6 +2292,30 @@ type NetworkSpec struct {
 	MultiClusterService MultiClusterServiceSpec `json:"multiClusterService,omitempty"`
 }
 
+// +kubebuilder:validation:Enum="";host;multus
+type NetworkProviderType string
+
+const (
+	NetworkProviderDefault = ""
+
+	NetworkProviderHost = "host"
+
+	NetworkProviderMultus = "multus"
+)
+
+type NetworkRangesSpec struct {
+	Public  []CIDR `json:"public"`
+	Cluster []CIDR `json:"cluster"`
+}
+
+// An IPv4 or IPv6 network CIDR.
+// +kubebuilder:validation:Pattern=`^[0-9a-fA-F:.]{2,}\/[0-9]{1,3}$`
```

Comment:
The pattern is this simple for ipv4 and v6 CIDRs? I imagined it would be more complex.

Reply:
Not even. The real pattern would be like 800 chars long or more. There is a comment below about the "naive" pattern

Reply:
I see, it provides some basic validation, but leaves the more complex validation to the code helpers